src/model/preemption.c

```diff
@@ -28,15 +28,14 @@ exception_t preemptionPoint(void)
      */
     if (ksWorkUnitsCompleted >= CONFIG_MAX_NUM_WORK_UNITS_PER_PREEMPTION) {
         ksWorkUnitsCompleted = 0;
-        if (isIRQPending()) {
-            return EXCEPTION_PREEMPTED;
 #ifdef CONFIG_KERNEL_MCS
-        } else {
-            updateTimestamp();
-            if (!(sc_active(NODE_STATE(ksCurSC)) && isSufficientAndSplittable()) || isCurDomainExpired()) {
-                return EXCEPTION_PREEMPTED;
-            }
+        updateTimestamp();
+        if (!(sc_active(NODE_STATE(ksCurSC)) && isSufficientAndSplittable()) || isCurDomainExpired()) {
+            return EXCEPTION_PREEMPTED;
+        } else
 #endif
+        if (isIRQPending()) {
```

Comment:
Now that mcsIRQ is called unconditionally when EXCEPTION_PREEMPTED occurs, we need to make sure that the timestamp has been updated incase IRQT_TO_IRQ(irq) != KERNEL_TIMER_IRQ

Reply:
Also the updateTimestamp in mcsIRQ can probably be removed.